### PR TITLE
fix(Flyout): WASM - Fixed build failing when attempting to create a f…

### PR DESCRIPTION
**Pull request by LHuss**
…lyout containing a control using a themeresource

GitHub Issue (If applicable): #4463 

## PR Type

What kind of change does this PR introduce?

- Bugfix
## What is the current behavior?

If a flyout contains a control that uses a themeresource, the build fails.

## What is the new behavior?

If a flyout contains a control that uses a themeresource, the build succeeds.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

File: src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.Reflection.cs [hunk 1]
```diff
@@ -129,28 +129,40 @@ private bool IsType(XamlType xamlType, ISymbol typeSymbol)
 		{
 			var type = FindType(xamlType);
 
-			if (type != null)
+			return IsType(type, typeSymbol);
+		}
+
+		private bool IsType(INamedTypeSymbol symbol, ISymbol typeSymbol)
```
Comment (davidjohnoliver) [inline src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlFileGenerator.Reflection.cs:135]: The parameter names here are a bit confusing (particularly compared to their respective types)

**Review by davidjohnoliver — APPROVED** [review]
This will unblock the use case. 

However I think that there's still a bug that these theme resource bindings (and probably ElementName bindings?) won't update correctly, because the update method is not getting hooked up. So we should probably keep the related issue open.

**Comment by jeromelaban** [comment]
> This will unblock the use case.
> 
> However I think that there's still a bug that these theme resource bindings (and probably ElementName bindings?) won't update correctly, because the update method is not getting hooked up. So we should probably keep the related issue open.

@davidjohnoliver in this case, because it's a flyout, the theme updating will likely not be impacted much because the flyout is likely to be closed and reopened. If it's an issue, let's open another one.

**Comment by jeromelaban** [comment]
UWP does not seem to like this:

```
src\SamplesApp\UITests.Shared\Windows_UI_Xaml_Controls\Flyout\Flyout_Local.xaml(9,5): Error WMC0011: Unknown member 'Background' on element 'Flyout'
```

**Review by jeromelaban — APPROVED** [review]

**Pull request merged**